sympy/physics/quantum/tests/test_tensorproduct.py

```diff
@@ -49,6 +49,11 @@ def test_tensor_product_simp():
     assert tensor_product_simp(TP(A, B)*TP(B, C)) == TP(A*B, B*C)
 
 
+def test_tensor_product_simp_Pow():
+    assert tensor_product_simp((TP(A, B)*TP(C, D)) ** 2) == (TP(A*C, B*D)) ** 2
+    assert tensor_product_simp(TP(A, B) ** 2) == TP(A**2, B**2)
```

Comment:
@johanbluecreek , I have added the test you mentioned previously.

Reply:
Test should cover all possible cases(`if` branches).

Reply:
The tests for all `if` branches have been implemented.
Please review, @ylemkimon .

Reply:
The tests should run `tensor_product_simp_Pow` instead of `tensor_product_simp`.

Reply:
I would also like to add that the first test, `assert tensor_product_simp((TP(A, B)*TP(C, D)) ** 2) == (TP(A*C, B*D)) ** 2`, is perhaps not appropriate. Since the "canonical form" (that is, applying `tensor_product_simp_Pow()` until the expression stops changing) would be
```
In []: tps(tps((tp(A, B)*tp(C, D)) ** 2))
Out[]: ((A*C)**2)x((B*D)**2)
```
Meaning if `tensor_product_simp_Pow()` is improved in the future to reach canonical form directly this test would fail for the wrong reason.